### ISSUE TITLE
ResourceGuard always results in 403 Forbidden Error?

### ISSUE THREAD

**Issue by IRuFFeYI** [ferrerojosh/nest-keycloak-connect#137]
I try to use Keycloak as  Authorization Services. I want to secure the endpoints of my REST API with resources, scopes and permissions.

The problem is, I get a 403 Forbidden Resource Error on every request. I try to explain my whole configuration with a lot of images. Am I missing something or have I misunderstood something?

## Keycloak Configuration

**Configuration of the client:**
![image](https://user-images.githubusercontent.com/34284916/195825191-70ddee93-2df5-401e-8470-2ea593c06761.png)

**My test rescource:**
![image](https://user-images.githubusercontent.com/34284916/195820249-c4326a9c-96cc-4ff1-bcb6-08049cee291c.png)

**My policy:**
![image](https://user-images.githubusercontent.com/34284916/195820351-e5761647-2f72-4ef9-97db-1002d56c2bd6.png)

**My permission:**
![image](https://user-images.githubusercontent.com/34284916/195820433-428b7546-423b-4e6e-8bef-adf4aced4f6c.png)

**Evaluation**
To test my configuration I use the internal evaluation tool.
I have the user: "_testcustomer_" who is member of the group: Testproject. Therefore he should have access to the resource. The evaluation shows, it's working correctly:
![image](https://user-images.githubusercontent.com/34284916/195821808-4088d8d2-e14e-4fdb-bd1e-69407588b49a.png)

## REST API:
In my NestJs REST API I'm using the _nest-keycloak-connect_-package. 

This is the configuration of the KeycloakModule:
![image](https://user-images.githubusercontent.com/34284916/195822708-f04cff92-9a59-488a-a920-d96fda7c1d3a.png)

**My Test-Controller**
![image](https://user-images.githubusercontent.com/34284916/195822889-b7ddbd08-db96-4a92-8371-80bfab48f3f3.png)

My understanding of the package:
When a user (in this case the testcustomer) sends a request to the API to the following URI: "http://localhost:3002/api/test/test", the resource from Keycloak is associated with it. And the testcustomer is authorized to call this endpoint. Other users who are not in the Testproject group will get a 403 Forbidden response.

The problem is that I always get a 403 error no matter if my user is in the group and it doesn't even matter if the resource exists in Keycloak at all. 

Note: General authentication or even authorization via roles works.

**Comment by ferrerojosh** [comment]
Ah I think its due to the fact your scopes have the `:` character in it.

If you think of the file in the official `keycloak-connect` library:
https://github.com/keycloak/keycloak-nodejs-connect/blob/f8e397cd7a1a763db5a45a07c3c3e40fec5cf8f2/middleware/enforcer.js#L18-L33

It splits off permission which is `permission:scope` by simply a split `:`.

What happened here is that the underyling `keycloak-connect` library receives a string of `File:files:read` and in gets sent to the keycloak enforcer as the resource `File` with the scope `files`.

tldr; the solution is to remove the `:` character in your scope. If it doesn't work out for you, you can submit a ticket to the `keycloak-connect` team as I do not have control over this behavior.

**Comment by IRuFFeYI** [comment]
Thank you very much. This really helped me so much. You made my week 😁
